earth_enterprise/src/server/wsgi/serve/publish/publish_manager_helper.py

```diff
@@ -1417,9 +1417,8 @@ def _GetVhNameToBaseUrlDict(self):
       (scheme://host[:port]) -  {vh_name: vh_base_url}
     """
     vh_list = self.QueryVhList()
-    vhname_to_baseurl_dct = {
-        vh_name: self.GetVhBaseUrl(vh_url, vh_ssl) for (
-            vh_name, vh_url, vh_ssl) in vh_list}
+    for (vh_name, vh_url, vh_ssl) in vh_list:
```

Comment:
Q: Have you tested this function? Do you not need to define `vhname_to_base_url_dict = {}` before you can assign its elements?

Reply:
those are virtual host names, I have not tested, but I corrected it as per your comments

Reply:
Q: I don't think the original code created a dictionary mapping keys to lists of `self.GetVhBaseUrl(. . .)` results.  It seemed just a map: key -> `self.GetVhBaseUrl(. . .)`.

Reply:
@tst-ppenev  sorry I am not sure I follow your comment.

Reply:
Try this for yourself.

File `p.py`:
```Python
class PublishManagerHelper(object):
  """Mock class with the _GetVhNameToBaseUrlDict() method on the
  'centos-6-support-new' branch."""

  def __init__(self, query_vh_list):
    self.query_vh_list = query_vh_list

  def QueryVhList(self):
    return self.query_vh_list

  def GetVhBaseUrl(self, vh_url, vh_ssl):
    return vh_url

  def _GetVhNameToBaseUrlDict(self):
    """Builds a dictionary mapping a virtual host name to a base URL.
    Returns:
      dictionary mapping a virtual host name to a base URL
      (scheme://host[:port]) -  {vh_name: vh_base_url}
    """
    vh_list = self.QueryVhList()
    vhname_to_baseurl_dct = dict(
        (vh_name,[]) for (vh_name, vh_url, vh_ssl) in vh_list)
    for (vh_name, vh_url, vh_ssl) in vh_list:
      vhname_to_baseurl_dct[vh_name].append(self.GetVhBaseUrl(vh_url, vh_ssl))

    return vhname_to_baseurl_dct
```

File `q.py`:
```Python
class PublishManagerHelper(object):
  """Mock class with the _GetVhNameToBaseUrlDict() method on the 'master'
  branch."""

  def __init__(self, query_vh_list):
    self.query_vh_list = query_vh_list

  def QueryVhList(self):
    return self.query_vh_list

  def GetVhBaseUrl(self, vh_url, vh_ssl):
    return vh_url

  def _GetVhNameToBaseUrlDict(self):
    """Builds a dictionary mapping a virtual host name to a base URL.
    Returns:
      dictionary mapping a virtual host name to a base URL
      (scheme://host[:port]) -  {vh_name: vh_base_url}
    """
    vh_list = self.QueryVhList()
    vhname_to_baseurl_dct = {
        vh_name: self.GetVhBaseUrl(vh_url, vh_ssl) for (
            vh_name, vh_url, vh_ssl) in vh_list}

    return vhname_to_baseurl_dct
```

Compare old method and updated method:
```Python
>>> import p
>>> import q
>>> u = p.PublishManagerHelper([('localhost', 'http://127.0.0.1', False), ('thanksgivinghost', 'https://bay.king.it', True)])
>>> u._GetVhNameToBaseUrlDict()
{'localhost': ['http://127.0.0.1'], 'thanksgivinghost': ['https://bay.king.it']}
# This method returns a dictionary with values which are lists.
>>> old = q.PublishManagerHelper([('localhost', 'http://127.0.0.1', False), ('thanksgivinghost', 'https://bay.king.it', True)])
>>> old._GetVhNameToBaseUrlDict()
{'localhost': 'http://127.0.0.1', 'thanksgivinghost': 'https://bay.king.it'}
# This method returns a dictionary with values which are not lists (strings in this case).
```